src/vs/workbench/contrib/tasks/common/problemMatcher.ts

```diff
@@ -1206,7 +1212,7 @@ class ProblemPatternRegistryImpl implements IProblemPatternRegistry {
 
 	private fillDefaults(): void {
 		this.add('msCompile', {
-			regexp: /^(?:\s+\d+>)?(\S.*)\((\d+|\d+,\d+|\d+,\d+,\d+,\d+)\)\s*:\s+(error|warning|info)\s+(\w+\d+)\s*:\s*(.*)$/,
+			regexp: /^(?:\s+\d+>)?(\S.*)\((\d+|\d+,\d+|\d+,\d+-\d+|\d+,\d+,\d+,\d+)\)\s*:\s+(error|warning|info)\s+(\w+\d+)\s*:\s*(.*)$/,
```

Comment:
This _could_ be done as `\d+,\d+(-\d+)?` too. Not sure which is preferable.

Reply:
Updated the regex to use capture groups instead of adding more alternations. I'm still not sure which is clearer.